pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -0,0 +1,207 @@
+		return l, err
+	}
+
+	accessorL.SetResourceVersion(accessorS.GetResourceVersion())
+	return l, nil
+}
+
+// List overrides the default behavior of the Storage and retrieves objects from
+// LegacyStorage or Storage depending on the DualWriter mode.
+func (d *DualWriterMode2) List(ctx context.Context, options *metainternalversion.ListOptions) (runtime.Object, error) {
+	legacy, ok := d.legacy.(rest.Lister)
+	if !ok {
+		return nil, fmt.Errorf("legacy storage rest.Lister is missing")
+	}
+
+	legacyList, err := legacy.List(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+	ll, err := meta.ExtractList(legacyList)
+	if err != nil {
+		return nil, err
+	}
+
+	storageList, err := d.Storage.List(ctx, options)
+	if err != nil {
+		return nil, err
+	}
+	sl, err := meta.ExtractList(storageList)
+	if err != nil {
+		return nil, err
+	}
+
+	type fieldsToEnrich struct {
+		resourceVersion string
+	}
+
+	m := map[string]fieldsToEnrich{}
+	for _, obj := range sl {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return nil, err
+		}
+		m[accessor.GetName()] = fieldsToEnrich{accessor.GetResourceVersion()}
+	}
+
+	for _, obj := range ll {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return nil, err
+		}
+		if entry, ok := m[accessor.GetName()]; ok {
+			accessor.SetResourceVersion(entry.resourceVersion)
+		}
+	}
+
+	if err = meta.SetList(legacyList, ll); err != nil {
+		return nil, err
+	}
+	return legacyList, nil
+}
+
+// Create overrides the default behavior of the Storage and writes to LegacyStorage and Storage depending on the dual writer mode.
+func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+	legacy, ok := d.legacy.(rest.Creater)
+	if !ok {
+		return nil, fmt.Errorf("legacy storage rest.Creater is missing")
+	}
+
+	created, err := legacy.Create(ctx, obj, createValidation, options)
+	if err != nil {
+		return created, err
+	}
+
+	accessor, err := meta.Accessor(created)
+	if err != nil {
+		return created, err
+	}
+	accessor.SetResourceVersion("")
+	accessor.SetUID("")
+
+	rsp, err := d.Storage.Create(ctx, created, createValidation, options)
+	if err != nil {
+		klog.Error("unable to create object in duplicate storage", "error", err, "mode", Mode2)
+	}
+	return rsp, err
+}
+
+// Update overrides the default behavior of the Storage and writes to both the LegacyStorage and Storage depending on the DualWriter mode.
+func (d *DualWriterMode2) Update(ctx context.Context, name string, objInfo rest.UpdatedObjectInfo, createValidation rest.ValidateObjectFunc, updateValidation rest.ValidateObjectUpdateFunc, forceAllowCreate bool, options *metav1.UpdateOptions) (runtime.Object, bool, error) {
+	legacy, ok := d.legacy.(rest.Updater)
+	if !ok {
+		return nil, false, fmt.Errorf("legacy storage rest.Updater is missing")
+	}
+
+	// Get the previous version from k8s storage (the one)
+	var old runtime.Object
```

Comment:
am I understanding correctly that `old` is defined again in the line below?

Reply:
At that point in the commit history I think that was the case yes.